### PR TITLE
PR1 - refactor post deletion

### PR DESCRIPTION
**Pull request by arminekhachatryan94**
Refactored post deletion to use a service and contract.

### CODE REVIEW

**Comment by kyledewey** [comment]
Review requested from @shanth2600

**Comment by kyledewey** [comment]
Additional review requested from @alejandrom2

File: forum/app/Services/PostService.php [hunk 1]
```diff
@@ -42,5 +43,32 @@ public function editPost($postData, $id){
             ], 401);
         }
     }
+
+    public function deletePost($request, $id){
+        $post = Post::find($id);
```
Comment (alejandrom2) [inline forum/app/Services/PostService.php]: I would recommend you used the findOrFail to help handle exception.

File: forum/app/Services/PostService.php [hunk 2]
```diff
@@ -42,5 +43,32 @@ public function editPost($postData, $id){
...
+                        'invalid' => 'You do not have permission to delete this post'
+                    ]
+                ], 401);
+            }
+        } else {
+            return response()->json([
+                'errors' => [
+                    'invalid' => 'Post does not exist'
+                ]
+            ], 401);
```
Comment (alejandrom2) [inline forum/app/Services/PostService.php]: Is 401 error the appropriate one for post not found?

**Review by shanth2600 — CHANGES_REQUESTED** [review]
I would relegate the access control (permissions) to its own service. Also the domain service (post service) shouldn't be dealing with json or HTTP responses. That's stuff that should happen at the infrastructure level (in the controller). ALSO inject the model into the service as a dependency and don't use the static '::' accessor for creating the sql queries. Slack me if you need clarification.

**Review by alejandrom2 — CHANGES_REQUESTED** [review]

**Review by alejandrom2 — APPROVED** [review]

**Review by shanth2600 — APPROVED** [review]

**Pull request merged**